modules/ds18b20/ds18b20.h

```diff
@@ -0,0 +1,315 @@
+#define DEVICE_NOT_DS18B20             0
+#define DS18B20_CONVERSION_SUCCESS     1
+#define DS18B20_CONVERSION_IN_PROGRESS 2
+#define DS18B20_CRC_INVALID            3
+
+/**
+ * @}
+ */
+
+/**
+ * @defgroup DS18B20_Typedefs
+ * @brief    Library Typedefs
+ * @{
+ */
+
+/**
+ * @brief  DS18B0 Resolutions available
+ */
+typedef enum {
+	DS18B20_Resolution_9bits  = 9,   /*!< DS18B20 9 bits resolution */
+	DS18B20_Resolution_10bits = 10, /*!< DS18B20 10 bits resolution */
+	DS18B20_Resolution_11bits = 11, /*!< DS18B20 11 bits resolution */
+	DS18B20_Resolution_12bits = 12  /*!< DS18B20 12 bits resolution */
+} DS18B20_Resolution_t;
+
+/**
+ * @}
+ */
+
+/**
+ * @defgroup DS18B20_Functions
+ * @brief    Library Functions
+ * @{
+ */
+
+/**
+ * @brief  Starts temperature conversion for specific DS18B20 on specific OneWireStruct channel
+ * @param  *OneWireStruct: Pointer to @ref OneWire_t working structure (OneWireStruct channel)
+ * @param  *ROM: Pointer to first byte of ROM address for desired DS12B80 device.
+ *         Entire ROM address is 8-bytes long
+ * @retval:
+ *          - DS18B20_USAGE_ERROR           
+ *          - DEVICE_NOT_DS18B20            
+ *          - 1 : Started DS18B20 at ROM address 
+ */
+char DS18B20_Start(OneWire_t* OneWireStruct, uint8_t* ROM);
+
+/**
+ * @brief  Starts temperature conversion for all DS18B20 devices on specific OneWireStruct channel
+ * @note   This mode will skip ROM addressing
+ * @param  *OneWireStruct: Pointer to @ref OneWire_t working structure (OneWireStruct channel)
+ * @retval:
+ *          - DS18B20_USAGE_ERROR           
+ *          - DEVICE_NOT_DS18B20            
+ *          - 1 : Started all DS18B20s on OneWire Bus   
+ */
+char DS18B20_StartAll(OneWire_t* OneWireStruct);
+
+/**
+ * @brief  Reads temperature from DS18B20
+ * @param  *OneWireStruct: Pointer to @ref OneWire_t working structure (OneWireStruct channel)
+ * @param  *ROM: Pointer to first byte of ROM address for desired DS12B80 device.
+ *         Entire ROM address is 8-bytes long
+ * @param  *destination: Pointer to float variable to store temperature
+ * @retval:
+ *          - DS18B20_USAGE_ERROR           
+ *          - DEVICE_NOT_DS18B20            
+ *          - DS18B20_CONVERSION_SUCCESS    
+ *          - DS18B20_CONVERSION_IN_PROGRESS
+ *          - DS18B20_CRC_INVALID           
+ */
+char DS18B20_Read(OneWire_t* OneWireStruct, uint8_t* ROM, float* destination);
+
+/**
+ * @brief  Gets resolution for temperature conversion from DS18B20 device
+ * @param  *OneWireStruct: Pointer to @ref OneWire_t working structure (OneWireStruct channel)
+ * @param  *ROM: Pointer to first byte of ROM address for desired DS12B80 device.
+ *         Entire ROM address is 8-bytes long
+ * @retval:
+ *          - DS18B20_USAGE_ERROR           
+ *          - DEVICE_NOT_DS18B20            
+ *          - 9 - 12: Resolution of DS18B20
+ */
+char DS18B20_GetResolution(OneWire_t* OneWireStruct, uint8_t* ROM);

```

Comment:
This seems odd to have a function return different types.  In this case it's either returning a status (error condition) or a desired value (9 - 12).  
I think it should return only 1 type, either status or the value.

If you want to have the status returned by the function, then return the desired value by reference:
char DS18B20_GetResolution(OneWire_t* OneWireStruct, uint8_t* ROM, uint8_t* resolution);

If you want to have the value returned by the function, then return the status by reference:
uint8_t DS18B20_GetResolution(OneWire_t* OneWireStruct, uint8_t* ROM, char* status);